src/python/pants/util/rwbuf.py

```diff
@@ -79,3 +79,29 @@ def __init__(self, backing_file):
 
   def do_write(self, s):
     self._io.write(s)
+
+
+class StringWriter(object):
+  """A write-only buffer which accepts strings and writes to another buffer which accepts bytes.
+
+  Writes strings as utf-8.
+
+  This is write-only because it's unclear whether seeking should seek by code-point or byte, and
+  implementing the former is non-trivial. If you need to read, read from the underlying buffer's
+  bytes.
+  """
+
+  def __init__(self, underlying):
+    # Called buffer to mirror how sys.stdout and sys.stderr expose this.
```

Comment:
Are you expecting `underlying` to be an `io.BytesIO`, `io.RawIOBase`, `io.BufferedIOBase`, etc? See https://docs.python.org/3/library/io.html#class-hierarchy.

A type hint (e.g. in the form of docstring, since we don't yet use inline type hints) would be very helpful for understanding this module.

FYI in Py3, `sys.stderr` is of type `io._TextIOWrapper` and `sys.stderr.buffer is `io.BufferedWriter`. In Python 2, it says it's an open file.

Reply:
Added a docstring; please let me know if there's a better way for me to describe this...

Reply:
Looks good. Once we have inline type hints, FYI it would look something like this

def init(underlying: io.BufferedWriter):

That is, we would have to identify the exact class it is. 

It’s okay to not do this now because it looks like the class changes between Py2 and Py3, and we don’t yet use type hints. This will all be easier once we drop Py2. 

(Sorry for syntax issues, on phone)